### PR TITLE
Streamline action inputs

### PR DESCRIPTION
**Pull request by thrix**
After reviewing I would a bit streamline the options, I believe they can
be shorter and grouped into different settings groups.

Also we should hide secrets even in case of debug output, safety first
:)

Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,39 +1,61 @@
...
+|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|-------------------------------|
+| `api_key`                    | Testing farm API key.                                                                                                             | empty, **required from user** |
+### Tmt Metadata
+
+| Input Name                   | Description                                                                                                                       | Default value                 |
+|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|-------------------------------|
+| `url`                        | An url to the repository with tmt metadata                                                                                        | empty, **required from user** |
```
Comment (zmiklank) [inline README.md:32]: I like the idea of logical groups, but in addition, I would suggest a prefix, based on them. This way they can be easily distinguishable even without the README.md

```suggestion
| `tmt_url`                        | An url to the repository with tmt metadata                                                                                        | empty, **required from user** |
```
The same can be applied to the "test environment" group with `secrets` as `test_secrets`, `variables` as `test_variables`, etc.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
+### Test Artifacts
+
 | `copr`                       | Copr name to use for the artifacts                                                                                                | epel-7-x86_64                 |
-| `test_artifacts`             | `fedora-copr-build` artifacts for testing environment, separated by ;                                                             | empty                         |
+| `copr_artifacts`             | `fedora-copr-build` artifacts for testing environment, separated by ;                                                             | empty                         |
+
+
+### Miscellaneous
+
+| `create_issue_comment`       | If GitHub action will create a github issue comment.                                                                              | false                         |
```
Comment (zmiklank) [inline README.md:55]: This is a question to @phracek:

Would `create_issue_comment`  be supported in further versions of testing-farm-as-a-github-action? It seems, that this variable is currently not used. If my assumption is true I would suggest removing this input, for now, so the user is not confused.

**Review by zmiklank — COMMENTED** [review]
Thank you for the PR. I wrote some notes, let me please know (and also @phracek) what do you think.

File: README.md (hunk 2 above)
Comment (phracek) [reply README.md:55]: I totally forgot to add this functionality.

I will do it as a separate pull request.

File: README.md [hunk 3]
```diff
@@ -1,39 +1,63 @@
...
-
 Before calling this GitHub Action, you must first clone your project,
 e.g., with the [Checkout V2](https://github.com/actions/checkout) GitHub Action.
 
-The Action uses ubuntu-20.04 and it is a [composite](https://docs.github.com/en/actions/creating-actions/about-custom-actions).
+The Action uses ubuntu-20.04 and it is a [composite action](https://docs.github.com/en/actions/creating-actions/about-custom-actions).
 It internally downloads needed binaries `curl` and `jq` for communicating with the Testing Farms API and parsing the responses.
  
-API key to the Testing Farm should be stored in your organization's secrets to successfully access its infrastructure.
+API key to the Testing Farm MUST be stored in your organization's secrets to successfully access its infrastructure.
```
Comment (phracek) [inline README.md:15]: It would be nice to have a link, where folks, can get the API key?

File: README.md [hunk 4]
```diff
@@ -1,39 +1,63 @@
...
+
+## Compatibility Notes
+
+⚠ Currently only testing with copr artifacts is supported.
```
Comment (phracek) [inline README.md:19]: It would be nice to have a description of what artifacts mean. Or link.

File: README.md (hunk 1 above)
Comment (phracek) [reply README.md:32]: URL is too generic. Some prefix is really nice. 
@thrix Do you have already the API v2 definition?

**Review by phracek — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (thrix) [reply README.md:32]: ack, ok, so git_url would be maybe better? :)

Comment (zmiklank) [reply README.md:32]: And is `tmt_url`, `tmt_plan`, `tmt_ref` and `tmt_context` instead of `url`, `plan`, `ref` and `context`  inconvenient?

Comment (thrix) [reply README.md:32]: @zmiklank it is not URL to tmt, it is a git repository URL same for ref, it is a git reference, others I agree

File: README.md (hunk 3 above)
Comment (thrix) [reply README.md:15]: I will add a comment, but we do not yet have onboarding guide on https://docs.testing-farm.io, let me draft something

File: README.md (hunk 1 above)
Comment (thrix) [reply README.md]: Asking for next round of review, sorry I forget how github does not like squasing :(

File: README.md (hunk 3 above)
Comment (thrix) [reply README.md:15]: I added link to the docs update which is being reviewed: https://gitlab.com/testing-farm/docs/root/-/merge_requests/3

File: README.md (hunk 4 above)
Comment (thrix) [reply README.md]: I added link, but we do not have it yet documented. I opened issue on our side: TFT-1041

File: README.md (hunk 2 above)
Comment (phracek) [reply README.md:56]: https://github.com/sclorg/testing-farm-as-github-action/pull/23

**Review by phracek — APPROVED** [review]
LGTM now. I would be glad for approve also from @zmiklank

**Comment by thrix** [comment]
@zmiklank yeah, the thing is this, we do not support only `tmt` but also `sti` as the test format in Testing Farm (I know the action does not at all care about this for now), might be other test formats will hit later. They have this in common, the git repository url and git ref is and will be always the main input. If we move it under `tmt_` prefix, then we would add support for `sti`, we would need add another set of vars `sti_git_url` and `sti_git_ref`. So I would like to generalize it. But if @phracek also thinks it will be better with prefix, I will obey :)

**Comment by zmiklank** [comment]
@thrix, now it makes more sense, and I agree with you. thanks for the clarification :).

**Review by zmiklank — APPROVED** [review]

**Comment by phracek** [comment]
As soon as it is rebased, we can merge it.

**Comment by phracek** [comment]
> Maybe one more question - are the `tmt_context` and `tmt_plan_regex` related only to the `tmt` then, and not also to the `sti` and other possible forms of declaring the test suites?

I guess, based on the API https://testing-farm.gitlab.io/api/, they are related to `tmt`, especially requirements.

**Comment by zmiklank** [comment]
@phracek, ok thanks.

**Comment by phracek** [comment]
I have rebased this pull request. @fernflower can you please briefly look at it? Maybe there are some changes that will not work in your workflow and will need some fixes in your repo.

**Pull request merged**